src/workers/cdc/primary.ts

```diff
@@ -740,6 +744,49 @@ const onPostRelationChange = async (
   }
 };
 
+const onMarketingCtaChange = async (
+  con: DataSource,
+  data: ChangeMessage<MarketingCta>,
+) => {
+  if (data.payload.op !== 'u') {
+    return;
+  }
+
+  const users = await con.getRepository(UserMarketingCta).findBy({
+    marketingCtaId: data.payload.after.campaignId,
+    readAt: IsNull(),
+  });
+
+  if (users.length > 0) {
+    await deleteRedisKey(
+      ...users.map((user) =>
+        generateStorageKey(
+          StorageTopic.Boot,
+          StorageKey.MarketingCta,
+          user.userId,
+        ),
+      ),
+    );
+  }
+};
+
+const onUserMarketingCtaChange = async (
+  data: ChangeMessage<UserMarketingCta>,
+) => {
+  // Only run on delete and if the user has read the Marketing CTA
+  if (data.payload.op !== 'd' || data.payload.before.readAt !== null) {
+    return;
+  }
```

Comment:
Can we split this into two separate returns?

Reply:
doen